### PR TITLE
Okina Makefile fix 

### PR DESCRIPTION
**Pull request by artv3**
Certain folders were not being created to store .o files. This PR modifies the makefile to create the folders (*/kernels).

### CODE REVIEW

File: makefile [hunk 1]
```diff
@@ -314,7 +314,8 @@ endif
 
 # Source dirs in logical order
 DIRS = general linalg mesh fem
-SOURCE_FILES  = $(foreach dir,$(DIRS),$(wildcard $(SRC)$(dir)/*.cpp))
+KERNEL_DIRS  = general/kernels linalg/kernels mesh/kernels fem/kernels
```
Comment (camierjs) [inline makefile]: Could DIRS be populated with $(dir)/kernels like the `SOURCE_FILES`? That would allow a one-line modification.

Comment (artv3) [reply makefile]: I don't think I quite follow. Doing something like 
TOP_DIRS = general linalg mesh fem
DIRS = $(TOP_DIRS) $(TOP_DIRS)/kernels
Doesn't quite work.

Comment (jdahm) [reply makefile]: I think he means
```
KERNEL_DIRS = $(DIRS:=/kernels)
```
... or `patsubst(%,%/kernels,$(DIRS))` works too.

Comment (artv3) [reply makefile]: I see, yup. Using ```KERNEL_DIRS = $(DIRS:=/kernels)`` does the job too 👍

**Pull request closed without merging**